packages/state-transition/src/block/processWithdrawals.ts

```diff
@@ -0,0 +1,68 @@
+import {ssz, capella} from "@lodestar/types";
+import {MAX_EFFECTIVE_BALANCE, ETH1_ADDRESS_WITHDRAWAL_PREFIX, MAX_WITHDRAWALS_PER_PAYLOAD} from "@lodestar/params";
+
+import {CachedBeaconStateCapella} from "../types.js";
+import {decreaseBalance} from "../util/index.js";
+
+export function getExpectedWithdrawals(state: CachedBeaconStateCapella): capella.Withdrawal[] {
+  const currentEpoch = state.epochCtx.epoch + 1;
+  let withdrawalIndex = state.nextWithdrawalIndex;
+  let validatorIndex = state.latestWithdrawalValidatorIndex;
+  const {validators, balances} = state;
+
+  const withdrawals: capella.Withdrawal[] = [];
+  for (let index = 0; index < validators.length; index++) {
+    // Get the index of validator next in turn
+    validatorIndex = (validatorIndex + 1) % validators.length;
+    const validator = validators.get(validatorIndex);
```

Comment:
it's important to use `validators.getReadonly(index)`, see https://github.com/ChainSafe/lodestar/pull/4404